### PR TITLE
Fix: Precission in VotingSettings

### PR DESCRIPTION
**Pull request by josemarinas**
## Description

Fix precission in Voting Setings

0.5 in sdk => 5x(10^17) in contracts

Task: [APP-1142](https://aragonassociation.atlassian.net/browse/APP-1142)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder of the package after the [UPCOMING] title and before the latest version.
- [ ] I have tested my code on the test network.

[APP-1142]: https://aragonassociation.atlassian.net/browse/APP-1142?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: modules/client/src/tokenVoting/internal/utils.ts [hunk 1]
```diff
@@ -79,21 +79,6 @@ export function toTokenVotingProposal(
...
-      // this should be decoded using the number of decimals that we want
-      // right now the encoders/recoders use 2 digit precission but the actual
-      // subgraph values are 18 digits precision. Uncomment below for 2 digits
-      // precision
-
-      // minSupport: decodeRatio(
-      //   BigInt(proposal.totalSupportThresholdPct),
-      //   2,
-      // ),
-      // minTurnout: decodeRatio(
```
Comment (brickpop) [inline modules/client/src/tokenVoting/internal/utils.ts]: If the solution comes later, removing this will make things a bit slower

**Review by brickpop — APPROVED** [review]

File: modules/client/src/tokenVoting/internal/utils.ts (hunk 1 above)
Comment (josemarinas) [reply modules/client/src/tokenVoting/internal/utils.ts]: My understanding is that this is not going to change, so I dont see a problem. 

The subgraph is settled in using 18 digits and we are ettled in using a number between 0 and 1

File: modules/client/src/addressList/internal/client/methods.ts [hunk 2]
```diff
@@ -436,10 +436,14 @@ export class ClientAddressListMethods extends ClientCore
       return {
         minDuration: parseInt(addresslistVotingPlugin.minDuration),
         supportThreshold: parseFloat(
-          formatEther(addresslistVotingPlugin.supportThreshold),
+          parseFloat(
+            formatEther(addresslistVotingPlugin.supportThreshold),
+          ).toFixed(2),
```
Comment (brickpop) [inline modules/client/src/addressList/internal/client/methods.ts]: Why don't we encapsulate this into a helper? This is verbose and spotting two parseFloat() together is weird

File: modules/client/src/tokenVoting/internal/utils.ts (hunk 1 above)
Comment (brickpop) [reply modules/client/src/tokenVoting/internal/utils.ts]: There is a task on the contract side to standardize the precision numbers (APP-1264).

Having 18 digits may make sense for token balances, but not to indicate a ratio

Comment (josemarinas) [reply modules/client/src/tokenVoting/internal/utils.ts]: Hmm, ok, nice to know then i'll leave this comment

File: modules/client/src/addressList/internal/client/methods.ts (hunk 2 above)
Comment (josemarinas) [reply modules/client/src/addressList/internal/client/methods.ts]: Sure, makes sense

File: modules/client/src/addressList/internal/client/methods.ts [hunk 3]
```diff
@@ -22,6 +22,7 @@ import {
   ClientCore,
   computeProposalStatusFilter,
   ContextPlugin,
+  etherToUnitInterval,
```
Comment (brickpop) [inline modules/client/src/addressList/internal/client/methods.ts]: interval? Something like `ratio` feels more meaningful. Maybe `parseEtherRatio()` or related?

File: modules/client/src/client-common/utils.ts [hunk 4]
```diff
@@ -110,3 +111,11 @@ export function votingModeToContracts(votingMode: VotingMode): number {
       throw new InvalidVotingModeError();
   }
 }
+
+export function etherToUnitInterval(ether: string): number {
```
Comment (brickpop) [inline modules/client/src/client-common/utils.ts]: how about parameterizing the amount of digits? 2 by default

File: modules/client/src/addressList/internal/client/methods.ts (hunk 3 above)
Comment (josemarinas) [reply modules/client/src/addressList/internal/client/methods.ts]: Yes i used unit interval bcause is the set of numbres between 0 and 1, but I wasnt super convinced about the name

**Review by brickpop — APPROVED** [review]

**Pull request merged**